File: src/Scripting/RubyRB/bridge.rb
class Device
  def initialize(name)
    @name = name
  end

  def turn_on = "#{@name}:on"
  def mute = "#{@name}:muted"
end

class Remote
  def initialize(device)
    @device = device
  end
end

class BasicRemote < Remote
  def execute = @device.turn_on
end

class MuteRemote < Remote
  def execute = @device.mute
end

tv = Device.new('TV')
radio = Device.new('Radio')
puts "basic-tv=#{BasicRemote.new(tv).execute}"
puts "basic-radio=#{BasicRemote.new(radio).execute}"
puts "mute-tv=#{MuteRemote.new(tv).execute}"
puts "mute-radio=#{MuteRemote.new(radio).execute}"

File: src/Web/JavaScriptJS/bridge.js
class TvDevice {
  powerOn() { return "TV:on"; }
  mute() { return "TV:muted"; }
}

class RadioDevice {
  powerOn() { return "Radio:on"; }
  mute() { return "Radio:muted"; }
}

class BasicRemote {
  constructor(device) { this.device = device; }
  activate() { return this.device.powerOn(); }
}

class MuteRemote {
  constructor(device) { this.device = device; }
  activate() { return this.device.mute(); }
}

const tv = new TvDevice();
const radio = new RadioDevice();
console.log(`basic-tv=${new BasicRemote(tv).activate()}`);
console.log(`basic-radio=${new BasicRemote(radio).activate()}`);
console.log(`mute-tv=${new MuteRemote(tv).activate()}`);
console.log(`mute-radio=${new MuteRemote(radio).activate()}`);
